shopping-cart/shopping-cart-java/shopping-cart/src/test/java/com/example/shoppingcart/impl/ShoppingCartReportTest.java

```diff
@@ -0,0 +1,123 @@
+package com.example.shoppingcart.impl;
+
+import com.lightbend.lagom.javadsl.persistence.Offset;
+import com.lightbend.lagom.javadsl.persistence.ReadSide;
+import com.lightbend.lagom.javadsl.persistence.jpa.JpaSession;
+import com.lightbend.lagom.javadsl.testkit.ReadSideTestDriver;
+import com.lightbend.lagom.javadsl.testkit.ServiceTest;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import akka.actor.ActorSystem;
+import akka.actor.ExtendedActorSystem;
+import akka.cluster.Cluster;
+
+import java.time.Instant;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.bind;
+import static com.lightbend.lagom.javadsl.testkit.ServiceTest.defaultSetup;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class ShoppingCartReportTest {
+
+    private final static ServiceTest.Setup setup = defaultSetup().withJdbc()
+            .configureBuilder(b ->
+                    b.overrides(
+                            bind(ReadSide.class).to(ReadSideTestDriver.class)
+                    )
+            );
+
+    private static ServiceTest.TestServer testServer;
+
+
+    private AtomicInteger offset;
+
+    @Before
+    public void restartOffset() {
+        offset = new AtomicInteger(1);
+    }
+
+    @BeforeClass
+    public static void beforeAll() {
+        testServer = ServiceTest.startServer(setup);
+
+        // Quickly form a Cluster with self: this is a workaround an issue in ReadSideTestDriver for Java
+        // see: https://github.com/lagom/lagom/issues/2140
+        ActorSystem actorSystem = testServer.injector().instanceOf(ActorSystem.class);
+        Cluster cluster = Cluster.get((ExtendedActorSystem) actorSystem);
+        cluster.join(cluster.selfAddress());
+    }
+
+    @AfterClass
+    public static void afterAll() {
+        if (testServer != null) testServer.stop();
+    }
+
+    private ReadSideTestDriver testDriver = testServer.injector().instanceOf(ReadSideTestDriver.class);
+    private ReportRepository reportRepository = testServer.injector().instanceOf(ReportRepository.class);
+
+    @Test @Ignore // ignored until https://github.com/lagom/lagom/issues/2140 is fixed
```

Comment:
@ignasi35, I said that those tests were passing locally but behaving flaky in Travis. I was wrong. They are also failing locally. 

Therefore, I'm disabling them for now. Until we solved https://github.com/lagom/lagom/issues/2140. 

We may want to merge this PR (even without the tests) because it introduces the read-side processor. That's the only added value. The testkit is not being properly demo'ed. 

Or we can leave the PR on hold and resume it after the 1.5.0 release.

Reply:
This is really annoying since `ReadSideTestDriver` [has been around for a while now](https://github.com/lagom/online-auction-java/search?q=ReadSideTestDriver&unscoped_q=ReadSideTestDriver). What do you think is the culprit?

> I notice, [...], that ReadSideTestDriver will wait for the registration of processor and that may fail if the cluster takes long to form.

Do you think moving to cluster bootstrap introduced a delay big enough for this issue to manifest? If so, then we _must_ fix lagom/lagom#2140 before lagom `1.6.0` GA.

PS: I noticed `ReadSideTestDriver` is not tested in lagom _itself_ probably the reason why we didn't notice the regression.

Reply:
That's a good guess. That would explain why we haven't seen it before.